Add operational excellence category to validate set of input

mudit794 opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

adxsdkps commented:
Can one of the admins verify this patch?

msJinLei commented:
Also need to change the markdown file of the cmdlet

msJinLei commented on src/Advisor/Advisor/ChangeLog.md:
```diff
@@ -19,6 +19,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Added new category "Operational Excellence"
```
Unclear change log. add new category to which cmdlet?

mudit794 commented:
> Also need to change the markdown file of the cmdlet

@msJinLei Updated markdown file.

> Also need to change the markdown file of the cmdlet

@msJinLei  addressed comments. Please have a look.

msJinLei commented on src/Advisor/Advisor/Cmdlets/Utilities/RecommendationHelper.cs:
```diff
@@ -51,8 +51,8 @@ public static List<PsAzureAdvisorResourceRecommendationBase> RecommendationFilte
-            // Filter by category only if its a valid input
-            if (Category.Cost.Equals(category) || Category.HighAvailability.Equals(category) || Category.Performance.Equals(category) || Category.Security.Equals(category))
+            // Filter by category
+            if (!string.IsNullOrEmpty(category))
```
the change breaks not test?

mudit794 replied:
It was redundant check. Input already checks valid category names. Simplified it.

msJinLei commented:
> > Also need to change the markdown file of the cmdlet
> 
> @msJinLei Updated markdown file.
> 
> > Also need to change the markdown file of the cmdlet
> 
> @msJinLei addressed comments. Please have a look.

Please see my comment
